zebra-chain/src/transaction/hash.rs

```diff
@@ -60,7 +60,7 @@ use super::{txid::TxIdBuilder, AuthDigest, Transaction};
 ///
 /// [ZIP-244]: https://zips.z.cash/zip-0244
 /// [Spec: Transaction Identifiers]: https://zips.z.cash/protocol/protocol.pdf#txnidentifiers
-#[derive(Copy, Clone, Eq, PartialEq, Serialize, Deserialize, Hash)]
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Serialize, Deserialize, Hash)]
```

Comment:
> Is the order here based on the display order, or the serialised order? (The byte order of the hash is reversed for display order.)

Looks like it's based on display order in zcashd.

> Transaction hashes don't have a natural order, so we might want to use a method that is named after the order we are providing. That would make reviews easier.
>
> What do you think?

I think that's a good idea and would be clearer.

Reply:
On second thought, it would be returning the same value as `mined_id()`, so I added a comment clarifying that we want the serialized byte order in [`25be33d`](https://github.com/ZcashFoundation/zebra/pull/5994/commits/25be33d9094bc671fef33e57a354f58214153f1e).